src/libraries/Microsoft.PowerFx.Json/Functions/JsonFunction.cs

```diff
@@ -0,0 +1,262 @@
+                    _flags.IgnoreUnsupportedTypes = arg1optionset.Option == "IgnoreUnsupportedTypes";
+                    _flags.IncludeBinaryData = arg1optionset.Option == "IncludeBinaryData";
+                    _flags.IndentFour = arg1optionset.Option == "IndentFour";
+                }
+
+                if ((_flags.IncludeBinaryData && _flags.IgnoreBinaryData) ||
+                    (_flags.HasMedia && !_flags.IncludeBinaryData && !_flags.IgnoreBinaryData) ||
+                    (!_flags.IgnoreUnsupportedTypes && HasUnsupportedType(_args[0].Type._type, out var _, out var _)))
+                {
+                    _flags = null;
+                    return;
+                }
+            }
+
+            internal void WriteToJson(Utf8JsonWriter writer, FormulaValue arg)
+            {
+                if (arg is BlankValue blankValue)
+                {
+                    writer.WriteNullValue();
+                }
+                else if (arg is BooleanValue booleanValue)
+                {
+                    writer.WriteBooleanValue(booleanValue.Value);
+                }
+                else if (arg is ColorValue colorValue)
+                {
+                    writer.WriteStringValue(GetColorString(colorValue.Value));
+                }
+                else if (arg is DateTimeValue dateTimeValue)
+                {
+                    writer.WriteStringValue(ConvertToUTC(dateTimeValue.GetConvertedValue(_timeZoneInfo), _timeZoneInfo).ToString("yyyy-MM-dd'T'HH:mm:ss.fffK", CultureInfo.InvariantCulture));
+                }
+                else if (arg is DateValue dateValue)
+                {
+                    writer.WriteStringValue(dateValue.GetConvertedValue(null).ToString("yyyy-MM-dd", CultureInfo.InvariantCulture));
+                }
+                else if (arg is DecimalValue decimalValue)
+                {
+                    writer.WriteNumberValue(decimalValue.Value);
+                }
+                else if (arg is GuidValue guidValue)
+                {
+                    writer.WriteStringValue(guidValue.Value.ToString("D", CultureInfo.InvariantCulture));
+                }
+                else if (arg is NumberValue numberValue)
+                {
+                    writer.WriteNumberValue(numberValue.Value);
+                }
+                else if (arg is OptionSetValue optionSetValue)
+                {
+                    if (optionSetValue.Type._type.OptionSetInfo is EnumSymbol enumSymbol)
+                    {
+                        if (enumSymbol.EntityName == "Color")
+                        {
+                            Color color = Color.FromArgb((int)(uint)(double)optionSetValue.ExecutionValue);
+                            writer.WriteStringValue(GetColorString(color));
+                        }
+                        else if (IsNumeric(optionSetValue.ExecutionValue, out object number))
```

Comment:
>IsNumeric( [](http://example.com/codeflow?start=33&length=10)

Can we avoid a double switch here and just do once.  
IsNumeric switches once, and then we switch again on 141...167 

And need some final else case for unhandled.

Reply:
Not sure of the value here
I will move the inner part inside IsNumeric if you prefer

No need to have a final else (IsNumeric would return false)